frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -10,74 +9,92 @@ import {
+  /* TBD(Afreen): Show installation storage class as preselected 
+                  Change state metadata
+  */
   const [inProgress, setProgress] = React.useState(false);
   const [errorMessage, setError] = React.useState('');
 
-  const osdSizeWithUnit = _.get(
-    ocsConfig,
-    'spec.storageDeviceSets[0].dataPVCTemplate.spec.resources.requests.storage',
-  );
-  const presentCount = _.get(ocsConfig, 'spec.storageDeviceSets[0].count');
-  const capacity = _.get(response, 'data.result[0].value[1]');
-
-  const osdSizeWithoutUnit = OSD_CAPACITY_SIZES[osdSizeWithUnit]?.size;
+  const cephCapacity: string = response?.data?.result?.[0]?.value[1];
+  const osdSizeWithUnit = getRequestedPVCSize(deviceSets[0].dataPVCTemplate);
+  const osdSizeWithoutUnit: number = OSD_CAPACITY_SIZES[osdSizeWithUnit]?.size;
   const provisionedCapacity = getProvisionedCapacity(osdSizeWithoutUnit);
+  const isNoProvionerSC: boolean = storageClass?.provisioner === NO_PROVISIONER;
+  const selectedSCName: string = getName(storageClass);
+  const deviceSetIndex: number = getCurrentDeviceSet(deviceSets, selectedSCName);
+  const { replica } = deviceSets[deviceSetIndex];
 
   let currentCapacity: React.ReactNode;
 
   if (loading) {
     currentCapacity = (
       <div className="skeleton-text ceph-add-capacity__current-capacity--loading" />
     );
-  } else if (loadError || !capacity || !presentCount || !osdSizeWithoutUnit) {
+  } else if (loadError || !cephCapacity || !osdSizeWithoutUnit || deviceSetIndex === -1) {
     currentCapacity = <div className="text-muted">Not available</div>;
   } else {
     currentCapacity = (
       <div className="text-muted">
-        <strong>{`${humanizeBinaryBytes(capacity / 3).string} / ${presentCount *
-          osdSizeWithoutUnit} TiB`}</strong>
+        <strong>{`${humanizeBinaryBytes(+cephCapacity / replica).string} / ${deviceSets[
+          deviceSetIndex
+        ].count * osdSizeWithoutUnit} TiB`}</strong>
       </div>
     );
   }
 
+  const onChange = (sc: StorageClassResourceKind) => setStorageClass(sc);
+
   const submit = (event: React.FormEvent<EventTarget>) => {
     event.preventDefault();
     setProgress(true);
-    const scName = getName(storageClass);
-    const patch = [
-      {
-        op: 'replace',
-        path: `/spec/storageDeviceSets/0/count`,
-        value: presentCount + 1,
-      },
-    ];
-    if (!scName) {
+    const patch = {
+      op: '',
+      path: '',
+      value: null,
+    };
+    const portable = !isNoProvionerSC;
+    const osdSize = isNoProvionerSC ? defaultRequestSize.BAREMETAL : osdSizeWithUnit;
+    if (deviceSetIndex === -1) {
+      patch.op = 'add';
+      patch.path = `/spec/storageDeviceSets/-`;
+      patch.value = createDeviceSet(selectedSCName, osdSize, portable);
+    } else {
+      patch.op = 'replace';
+      patch.path = `/spec/storageDeviceSets/${deviceSetIndex}/count`;
+      patch.value = deviceSets[deviceSetIndex].count + 1;
+    }
```

Comment:
```suggestion
    const portable = !isNoProvionerSC;
    const osdSize = isNoProvionerSC ? defaultRequestSize.BAREMETAL : osdSizeWithUnit;
   const patch = deviceSetIndex === -1 ?  {
      op : 'add';
      path : `/spec/storageDeviceSets/-`;
      value : createDeviceSet(selectedSCName, osdSize, portable);
    : {
      op : 'replace';
      path : `/spec/storageDeviceSets/${deviceSetIndex}/count`;
      value : deviceSets[deviceSetIndex].count + 1;
    }
```

Reply:
design preferences :)

Reply:
I prefer we not mutating objects when we have already defined them as `const`.

Reply:
Change will happen once in each if -else. I would prefer to keep it.
Thanks for suggesting